openedx/core/djangoapps/content_libraries/libraries_index.py

```diff
@@ -242,17 +243,16 @@ def get_item_definition(cls, item):
 
 @receiver(CONTENT_LIBRARY_CREATED)
 @receiver(CONTENT_LIBRARY_UPDATED)
-@receiver(LIBRARY_BLOCK_CREATED)
-@receiver(LIBRARY_BLOCK_UPDATED)
-@receiver(LIBRARY_BLOCK_DELETED)
-def index_library(sender, library_key, **kwargs):  # pylint: disable=unused-argument
+def index_library(sender, content_library: ContentLibraryData, **kwargs):  # pylint: disable=unused-argument
```

Comment:
@pomegranited The change in the event call affects here. If there are other receivers, they will break. I searched through the repositories and found none, but I'm not sure.

Reply:
These are such new signals, I'm pretty sure that changing the params is fine. It's worth it to migrate these events into openedx-events, bringing them in line with [OEP-50](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0050-hooks-extension-framework.html), and making it possible to send these events via the event bus if somebody wants them.

Could you please mark your commits with `feat!: `? (note the exclamation point) That flags breaking changes which can be highlighted during release preparation (cf [OEP-51 Types](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0051-bp-conventional-commits.html#type)). We can also post on https://discuss.openedx.org to gazette it if anyone's concerned.

@bradenmacdonald CC @arbrandes Do you have any concerns about us changing how these content library signals are sent and received, so long as content libraries and their tests continue to work?

Reply:
I think it's fine as libraries v2 aren't widely used yet, and this is a somewhat unstable API. The more stable events are in https://github.com/openedx/openedx-events anyways (should we be using those type of events?)

But if this were a stable API, I'd say to leave it as is and push the new data through the `**kwargs` - isn't that what they're for?

Reply:
Agree with @bradenmacdonald . You can push the data through `**kwargs`. [COURSE_CATALOG_INFO_CHANGED handler example](https://github.com/openedx/course-discovery/blob/a948a741b1839e845e07c762bab6ee51622daea5/course_discovery/apps/course_metadata/signals.py#L334-L346). Then we can also remove unnecessary pylint annotation.

Reply:
Hi @navinkarkera!
I think that using `**kwargs` for Open edX event data breaks [this](https://github.com/openedx/edx-platform/pull/32599#discussion_r1256401572) convention. The discussion (related to a similar, but another context) went further [here](https://github.com/openedx/edx-platform/pull/32599#discussion_r1256401572).

Reply:
@rpenido Sorry If I was not clear before. I am suggesting to change the signal handler definition to take in just `**kwargs` like the [COURSE_CATALOG_INFO_CHANGED handler example](https://github.com/openedx/course-discovery/blob/a948a741b1839e845e07c762bab6ee51622daea5/course_discovery/apps/course_metadata/signals.py#L334-L346). The discussion that you pointed to seems to be about adding fields to event data like `version`.

Reply:
Now I got it! Will do tomorrow!
Thank you!

Reply:
Done: 281f8b67b482caf3b007dc3abaeb7f02f6256a56